copyListWithRandomPointer.py

```diff
@@ -0,0 +1,37 @@
+# Definition for singly-linked list with a random pointer.
+# class RandomListNode(object):
+#     def __init__(self, x):
+#         self.label = x
+#         self.next = None
+#         self.random = None
+
+class Solution(object):
+    def copyRandomList(self, head):
+        """
+        :type head: RandomListNode
+        :rtype: RandomListNode
+        """
+        if not head:
+            return None
+        newHead = RandomListNode(head.label)
+        summary = {}
```

Comment:
Not sure if this is counted as using extra space.
Maybe worth looking at https://github.com/nkukarl/leetcode/blob/master/copy_list_with_random_pointer.py